packages/leancode_contracts_generator/bin/leancode_contracts_generator.dart

```diff
@@ -1,81 +1,85 @@
-      final configFile = File(configFileName);
-      if (!configFile.existsSync()) {
-        fatalError(
-          'No $configFileName config file found in the current directory. Did you run `${runner.executableName} ${initCommand.name}`?',
-        );
-      }
-
-      final generator = ContractsGenerator(
-        ContractsGeneratorConfig.fromYaml(await configFile.readAsString()),
-      );
-
-      await generator.writeAll();
-      exit(0);
-    }
-
-    await runner.run(arguments);
+    final statusCode = await runner.run(arguments) ?? 0;
+    exit(statusCode);
   } on ContractsGeneratorException catch (err) {
-    fatalError(err.message);
+    printError(err.message);
   } catch (err, st) {
     stderr
       ..writeln(err)
       ..writeln(st);
-    fatalError(
+    printError(
       'An unknown error occurred, this is most likely a bug. '
       'Please consider reporting it with the above error.',
     );
   }
+
+  exit(1);
 }
 
-class InitCommand extends Command<void> {
-  @override
-  final name = 'init';
-  @override
-  final description = 'Initialize a config file';
+class GeneratorCommandRunner extends CommandRunner<int> {
+  GeneratorCommandRunner()
+      : _initCommand = InitCommand(),
+        super(
+          'leancode_contracts_generator',
+          'Command line utility for generating dart CQRS contracts',
+        ) {
+    addCommand(_initCommand);
 
-  @override
-  String get invocation {
-    final parents = [name];
-    for (var command = parent; command != null; command = command.parent) {
-      parents.add(command.name);
-    }
-    parents.add(runner!.executableName);
-
-    final invocation = parents.reversed.join(' ');
-    return '$invocation [directory (default: .)]';
+    argParser.addFlag(
+      verboseFlag,
+      abbr: 'v',
+      help: 'Increase logging.',
+      negatable: false,
+    );
   }
 
+  final InitCommand _initCommand;
+
   @override
-  Future<void> run() async {
-    final dir = argResults!.rest.firstOrNull ?? '.';
+  Future<int?> run(Iterable<String> args) async {
+    final parsedArgs = argParser.parse(args);
+    final verbose = parsedArgs.verbose;
 
-    await File(p.join(dir, configFileName))
-        .writeAsString(ContractsGeneratorConfig.defaultYamlConfig);
+    if (parsedArgs.command == null) {
+      final configFile = File(configFileName);
+      if (!configFile.existsSync()) {
+        printError(
+          'No $configFileName config file found in the current directory. Did you run `$executableName ${_initCommand.name}`?',
+        );
+        return 1;
+      }
 
-    exit(0);
+      final generator = ContractsGenerator(
+        ContractsGeneratorConfig.fromYaml(await configFile.readAsString()),
+        verbose: verbose,
+      );
+
+      await generator.writeAll();
+      return 0;
+    }
+
+    return super.run(args);
   }
 }
 
-Never fatalError(String message) {
+void printError(String message) {
   stderr.writeln('\x1B[31m$message\x1B[0m');
```

Comment:
Maybe use [package:ansi_styles](https://pub.dev/packages/ansi_styles)?

Reply:
If I need more styles I will most likely use this package. But I don't do any logging other than verbose and error, so for now I don't need it